components/higher-order/with-dragging/index.js

```diff
@@ -0,0 +1,163 @@
+ */
+import './style.scss';
+
+const withDragging = ( OriginalComponent ) => {
+	class Draggable extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.onDragStart = this.onDragStart.bind( this );
+			this.onDragOver = this.onDragOver.bind( this );
+			this.onDragEnd = this.onDragEnd.bind( this );
+			this.cursorTop = null;
+			this.cursorLeft = null;
+			this.config = {
+				dragImageClass: 'components-with-dragging__invisible-drag-image',
+				cloneWrapperClass: 'components-with-dragging__clone',
+				cloneHeightTransformationBreakpoint: 700,
+				clonePadding: 20,
+			};
+			this.state = {
+				cloneNodeId: null,
+				elementId: null,
+			};
+		}
+
+		/**
+		 * Removes the element clone, resets cursor, and removes drag listener.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 */
+		onDragEnd( event ) {
+			const { elementId, cloneNodeId } = this.state;
+			const element = document.getElementById( elementId );
+			const cloneWrapper = document.getElementById( cloneNodeId );
+
+			if ( element && cloneWrapper ) {
+				// Remove clone.
+				element.parentNode.removeChild( cloneWrapper );
+			}
+
+			// Reset cursor.
+			document.body.classList.remove( 'dragging' );
+
+			this.setState( { elementId: null, cloneNodeId: null } );
+
+			document.removeEventListener( 'dragover', this.onDragOver );
+			event.stopPropagation();
+		}
+
+		/*
+		 * Updates positioning of element clone based on mouse movement during dragging.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 */
+		onDragOver( event ) {
+			const { cloneNodeId } = this.state;
+			const cloneWrapper = document.getElementById( cloneNodeId );
+
+			cloneWrapper.style.top =
+				`${ parseInt( cloneWrapper.style.top, 10 ) + parseInt( event.clientY, 10 ) - parseInt( this.cursorTop, 10 ) }px`;
+			cloneWrapper.style.left =
+				`${ parseInt( cloneWrapper.style.left, 10 ) + parseInt( event.clientX, 10 ) - parseInt( this.cursorLeft, 10 ) }px`;
+
+			// Update cursor coordinates.
+			this.cursorLeft = event.clientX;
+			this.cursorTop = event.clientY;
+		}
+
+		/**
+		 *  - Clones the current element and spawns clone over original element.
+		 *  - Adds a fake temporary drag image to avoid browser defaults.
+		 *  - Sets transfer data.
+		 *  - Adds dragover listener.
+		 * @param  {Object} event     The non-custom DragEvent.
+		 * @param  {string} uid       The unique identifier of the element to be dragged.
+		 * @param  {string} elementId The HTML id of the element to be dragged.
+		 * @param  {number} order     The current index of the element to be dragged.
+		 * @param  {string} type      A unique constant identifying the type of dragging operation.
+		 */
+		onDragStart( event, uid, elementId, order, type ) {
+			const element = document.getElementById( elementId );
+			const elementWrapper = element.parentNode;
+			const elementRect = element.getBoundingClientRect();
+			const elementTopOffset = parseInt( elementRect.top, 10 );
```

Comment:
Is `parseInt` necessary?

Reply:
I remember these parsers were necessary - just can't remember right now which browser was causing me trouble. I will investigate. See also my comment above on `clientY` etc.